testkit/src/main/java/com/akkaserverless/javasdk/testkit/ValueEntityResult.java

```diff
@@ -20,23 +20,43 @@
  * Represents the result of an ValueEntity handling a command when run in through the testkit.
  *
  * @param <R> The type of reply that is expected from invoking command handler
+ *     <p>Not for user extension, returned by the generated testkit.
  */
-// FIXME a way to inspect other effects than reply?
-public final class ValueEntityResult<R> {
+public interface ValueEntityResult<R> {
 
-  private final R reply;
+  /** @return true if the call had an effect with a reply, false if not */
+  boolean isReply();
 
   /**
-   * INTERNAL API
-   *
-   * <p>Constructed by the generated code, not intended for calls from user code.
+   * The reply object from the handler if there was one. If the call had an effect without any reply
+   * an exception is thrown
    */
-  public ValueEntityResult(R reply) {
-    this.reply = reply;
-  }
-
-  /** The reply object from the handler if there was one. */
-  public R getReply() {
-    return reply;
-  }
+  R getReply();
+
+  /** @return true if the call was forwarded, false if not */
+  boolean isForward();
+
+  /**
+   * An object with details about the forward. If the result was not a forward an exception is
+   * thrown
+   */
+  ServiceCallDetails<R> getForward();
+
+  /** @return true if the call was an error, false if not */
+  boolean isError();
+
+  /** The error description. If the result was not an error an exception is thrown */
+  String getError();
+
+  /** @return true if the call had a noReply effect, false if not */
+  boolean isNoReply();
+
+  /** @return true if the call updated the entity state */
+  boolean stateWasUpdated();
```

Comment:
Since this is asking a question, I think it will read better if: `wasStateUpdated()`

Reply:
I started renaming, but after seeing it in use I think i disagree: `if (result.stateWasUpdated()) / assertTrue(result.stateWasUpdated())` vs `if (result.wasStateUpdated()) / assertTrue(result.wasStateUpdated())`

current reads better, no?

Reply:
Ok, fine by me